core/src/main/java/org/apache/calcite/prepare/CalciteCatalogReader.java

```diff
@@ -368,6 +366,20 @@ private static SqlOperator toOp(SqlIdentifier name,
     }
   }
 
+  private static SqlSyntax getSqlSyntax(org.apache.calcite.schema.Function function,
+      CalciteConnectionConfig config) {
+    if (!function.getParameters().isEmpty()) {
+      return SqlSyntax.FUNCTION;
+    }
+    // Keep compatible with both Foo() and Foo function syntax for Calcite's default conformance
```

Comment:
Does this change only affect UDF behavior?

Reply:
Yes, it only affects user-defined UDF functions, not other built-in functions.